### ISSUE TITLE
🐛 <VSCode> When saving tsconfig.json, biome extension reports an error msg.

### ISSUE THREAD

**Issue by zhoujian-official** [biomejs/biome#1055]
### Environment information

```block
CLI:
  Version:                      1.4.1
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.10.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/10.2.3"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### What happened?

1. Enable editor.formatOnSave.
2. Open tsconfig.json in vscode.
3. Edit it and save.
4. Biome extension shows an error message: Format with errors is disabled.

![image](https://github.com/biomejs/biome/assets/35226412/d1b4fc99-07bd-4c8e-a041-f451925c109b)

I think it is related to #933 

### Expected result

1. don't show this error
2. treat tsconfig.json as a "json with comments" file like vscode did, and format it.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
If you have the chance, could you try the latest nightly release? Just to confirm that it's fixed

**Comment by ematipico** [comment]
> > Is the bug still present in the latest release (1.5.2)?
> 
> The error message is gone, but `tsconfig.json ` is not formatted, is this expected?

That's expected. It's a protected file: https://biomejs.dev/guides/how-biome-works/#protected-files

**Comment by kevinwolfcr** [comment]
@ematipico Is there any specific reason Biome can not format `tsconfig.json` files? I would love to apply my formatter rules to that file, so it follows the same style as the rest of the json files on my projects. I see that the latest [nightly release](https://github.com/biomejs/biome/releases/tag/cli%2Fv1.5.3-nightly.6cbf1c7) allows formatting `package.json`.

**Comment by ematipico** [comment]
> @ematipico Is there any specific reason Biome can not format `tsconfig.json` files? I would love to apply my formatter rules to that file, so it follows the same style as the rest of the json files on my projects. I see that the latest [nightly release](https://github.com/biomejs/biome/releases/tag/cli%2Fv1.5.3-nightly.6cbf1c7) allows formatting `package.json`.

Because it requires special formatting. Would you like to help and remove the restrictions?

**Comment by kevinwolfcr** [comment]
>  Because it requires special formatting. Would you like to help and remove the restrictions?

Sure I can help. I asume that it is just allowing TSconfig to be linted and formatted, right? This will be my first time doing something in rust, but I will take a look at the PR enabling `package.json`.

What is the special formatting you mean?

**Comment by ematipico** [comment]
> > Because it requires special formatting. Would you like to help and remove the restrictions?
> 
> Sure I can help. I asume that it is just allowing TSconfig to be linted and formatted, right? This will be my first time doing something in rust, but I will take a look at the PR enabling `package.json`.
> 
> What is the special formatting you mean?

`tsconfig.json` and `jsconfig.json` must be parsed as JSONC files, because that's why `tsc` and VSCode parse these files. I order to opt-in this, you'll have to add these files in this array:

https://github.com/biomejs/biome/blob/6560177be8d5c89ca1b84c2cbdbba6cc05f8de57/crates/biome_service/src/file_handlers/mod.rs#L65-L78
